lam4doc/config.py

```diff
@@ -8,40 +8,138 @@
 """
 Project wide configuration file.
 """
-
+import logging
 import os
 from pathlib import Path
 
-LAM_DEBUG = os.environ.get('LAM_DEBUG', True)
-LAM_FLASK_SECRET_KEY = os.environ.get('LAM_FLASK_SECRET_KEY', 'secret key')
-
-LAM_API_LOCATION = os.environ.get('LAM_API_LOCATION', 'http://lam-api')
-LAM_API_PORT = os.environ.get('LAM_API_PORT', 4050)
-LAM_API_SERVICE = f'{LAM_API_LOCATION}:{LAM_API_PORT}'
-
-LAM_FUSEKI_LOCATION = os.environ.get('LAM_FUSEKI_LOCATION', 'http://fuseki')
-LAM_FUSEKI_PORT = os.environ.get('LAM_FUSEKI_PORT', 3030)
-LAM_FUSEKI_QUERY_URL = os.environ.get('LAM_FUSEKI_QUERY_URL', '/lam/query')
-LAM_FUSEKI_SERVICE = f'{LAM_FUSEKI_LOCATION}:{LAM_FUSEKI_PORT}{LAM_FUSEKI_QUERY_URL}'
-LAM_FUSEKI_USERNAME = os.environ.get('LAM_FUSEKI_USERNAME', 'admin')
-LAM_FUSEKI_PASSWORD = os.environ.get('LAM_FUSEKI_PASSWORD', 'admin')
-
-LAM_REPORT_TEMPLATE_LOCATION = str(Path(__file__).parents[1] / 'templates/content')
-LAM_REPORT_CONFIG = str(Path(__file__).parents[1] / 'templates/content/config.json')
-
-LAM_INDEXES_TEMPLATE_LOCATION = str(Path(__file__).parents[1] / 'templates/indexes')
-LAM_CELEX_CONFIG_NAME = 'celex.json'
-LAM_CLASSES_CONFIG_NAME = 'classes.json'
-LAM_PROPERTIES_CONFIG_NAME = 'properties.json'
-
-LAM_DEFAULT_TIMEOUT = int(os.environ.get('LAM_GUNICORN_TIMEOUT', 300))
-
-LAM_DOCUMENT_PROPERTY_GRAPH = os.environ.get("LAM_DOCUMENT_PROPERTY_GRAPH", "http://publications.europa.eu/resources/authority/lam/DocumentProperty")
-LAM_CLASSES_GRAPH = os.environ.get("LAM_CLASSES_GRAPH", "http://publications.europa.eu/resources/authority/lam/LAMLegalDocument")
-LAM_CELEX_CLASSES_GRAPH = os.environ.get("LAM_CELEX_CLASSES_GRAPH", "http://publications.europa.eu/resources/authority/celex/CelexLegalDocument")
-
-
-LAM_LOGGER = 'lam'
+HTML_REPORT_TYPE = 'html'
+PDF_REPORT_TYPE = 'pdf'
+DEFAULT_REPORT_TYPE = HTML_REPORT_TYPE
+REPORT_EXTENSIONS = [HTML_REPORT_TYPE, PDF_REPORT_TYPE]
+
+
+class LAMConfig:
```

Comment:
What we can do is instantiate this class as a global, **singleton**, module-level variable. 
Thienw e do not need to import and instantiate it in each module, you will end up only importing.